tests/particle_count_statistics/statistics

```diff
@@ -0,0 +1,29 @@
+# 1: Time step number
+# 2: Time (seconds)
+# 3: Number of mesh cells
+# 4: Number of Stokes degrees of freedom
+# 5: Number of temperature degrees of freedom
+# 6: Iterations for Stokes solver
+# 7: Velocity iterations in Stokes preconditioner
+# 8: Schur complement iterations in Stokes preconditioner
+# 9: Time step size (seconds)
+# 10: Number of advected particles
+# 11: Minimal particles per cell: 
+# 12: Average particles per cell: 
+# 13: Maximal particles per cell: 
+ 0 0.000000000000e+00 256 2467 1089 34 41   5 0.000000000000e+00    0 0 0  0 
+ 1 8.808144398380e+00 256 2467 1089  0  0   0 8.808144398380e+00 1000 0 3 11 
+ 2 1.761628879676e+01 256 2467 1089  0  0   0 8.808144398380e+00 1000 0 3 11 
+ 3 2.642443319514e+01 256 2467 1089  0  0   0 8.808144398380e+00 1000 0 3 11 
+ 4 3.523257759352e+01 256 2467 1089  0  0   0 8.808144398380e+00 1000 0 3 11 
+ 5 4.404072199190e+01 256 2467 1089  0  0   0 8.808144398380e+00 1000 0 3 11 
+ 6 5.284886639028e+01 256 2467 1089  0  0   0 8.808144398380e+00 1000 0 3 11 
+ 7 6.165701078866e+01 256 2467 1089  0  0   0 8.808144398380e+00 1000 0 3 11 
+ 8 7.046515518704e+01 256 2467 1089  0  0   0 8.808144398380e+00 1000 0 3 11 
+ 9 7.927329958542e+01 256 2467 1089  0  0   0 8.808144398380e+00 1000 0 3 11 
+10 8.808144398380e+01 256 2467 1089  0  0   0 8.808144398380e+00 1000 0 3 12 
+11 9.688958838218e+01 238 2344 1035  0  0   0 8.808144398380e+00 1000 0 3 11 
+ 0 0.000000000000e+00   0    0    0 29 30 120 0.000000000000e+00    0 0 0  0 
+12 1.000000000000e+02 238 2344 1035  0  0   0 3.110411617823e+00 1000 0 4 16 
+ 0 0.000000000000e+00   0    0    0 27 28 112 0.000000000000e+00    0 0 0  0 
```

Comment:
I guess, this is what is happening if one selects a nonlinear solver, and the linear solver tolerance is fulfilled by the initial guess in a timestep (the line of timestep 11), but the nonlinear tolerance is not. I am not sure why the linear solver then tries to solve something in the second nonlinear iteration (the lines with 0), because its residual should still be lower than the tolerance, but this behaviour is identical to the sol_cx test, so it is not influenced by this PR.